test/problems/sampling/protein_folding/bead_contacts/test_contact_map_builder.py

```diff
@@ -0,0 +1,206 @@
+        peptide = Peptide(
+            main_chain_len, main_chain_residue_seq, side_chains, side_chain_residue_sequences
+        )
+        (
+            lower_main_upper_main,
+            lower_side_upper_main,
+            lower_main_upper_side,
+            lower_side_upper_side,
+            r_contact,
+        ) = contact_map_builder._create_contact_qubits(peptide)
+
+        assert lower_main_upper_main == {}
+        assert lower_side_upper_main == {}
+        assert lower_main_upper_side == {}
+        assert lower_side_upper_side == {}
+        assert r_contact == 0
+
+    def test_create_pauli_for_contacts_2(self):
+        """
+        Tests that Pauli operators for contact qubits are created correctly.
+        """
+        main_chain_residue_seq = ["S", "A", "A", "S", "S", "S"]
+        main_chain_len = 6
+        side_chains = [0, 0, 1, 1, 1, 0]
+        side_chain_residue_sequences = [None, None, "A", "A", "S", None]
+        peptide = Peptide(
+            main_chain_len, main_chain_residue_seq, side_chains, side_chain_residue_sequences
+        )
+        (
+            lower_main_upper_main,
+            lower_side_upper_main,
+            lower_main_upper_side,
+            lower_side_upper_side,
+            r_contact,
+        ) = contact_map_builder._create_contact_qubits(peptide)
+
+        assert lower_main_upper_main == {
+            1: {
+                6: 0.5
+                * (I ^ I ^ I ^ I ^ I ^ I ^ I ^ I ^ I ^ I ^ I ^ I ^ I ^ I ^ I ^ I ^ I ^ I ^ I ^ I)
+                - 0.5
+                * (I ^ I ^ I ^ I ^ I ^ I ^ I ^ I ^ I ^ I ^ I ^ I ^ I ^ I ^ Z ^ I ^ I ^ I ^ I ^ Z)
+            }
+        }
+        assert lower_side_upper_main == {
+            1: {
+                5: 0.5
+                * (I ^ I ^ I ^ I ^ I ^ I ^ I ^ I ^ I ^ I ^ I ^ I ^ I ^ I ^ I ^ I ^ I ^ I ^ I ^ I)
+                - 0.5
+                * (I ^ I ^ I ^ I ^ I ^ Z ^ I ^ I ^ I ^ I ^ I ^ I ^ I ^ I ^ I ^ I ^ I ^ I ^ I ^ Z)
+            }
+        }
+        assert lower_main_upper_side == {}
+        assert lower_side_upper_side == {}
+        assert r_contact == 2
+
+    def test_create_pauli_for_contacts_3(self):
+        """
+        Tests that Pauli operators for contact qubits are created correctly.
+        """
+        main_chain_residue_seq = ["S", "A", "A", "S", "S", "S", "S"]
+        main_chain_len = 7
+        side_chains = [0, 0, 1, 1, 1, 1, 0]
+        side_chain_residue_sequences = [None, None, "A", "A", "S", "A", None]
+        peptide = Peptide(
+            main_chain_len, main_chain_residue_seq, side_chains, side_chain_residue_sequences
+        )
+        (
+            lower_main_upper_main,
+            lower_side_upper_main,
+            lower_main_upper_side,
+            lower_side_upper_side,
+            r_contact,
+        ) = contact_map_builder._create_contact_qubits(peptide)
+        expected_path = self.get_resource_path(
+            "test_create_pauli_for_contacts_3_expected_1",
```

Comment:
This is a general comment. You should not write expected case in files. The reasons are

1. output of `__str__` may be changed in the future.
2. file loading cost.

Instead, you should write
```python
PauliSumOp.from_list([("IIIIIIIIIIIIIIIIIIIIIIII", 0.5), ("IIIIIIIIIIIIIIIIIIZIIIIZ", -0.5)])
```
here.

And you can use `assertEqual` instead of `assert`.

Reply:
Tbh I don't agree. While the example you give is very short, some other of our test cases are **huge**. Especially with black formatting that would result in test files exceeding 1000 lines of code, making reviewing those test files unbearable..

Reply:
Is a huge unit test really a unit test? I'm concerned that you are trying to do integration test.

Even if putting test cases in external files is allowed, comparing them as strings is very dangerous.
We are planning to change the string https://github.com/Qiskit/qiskit-terra/pull/6081.

For example, I recommend using CSV or TSV or json format, and then parse and load it to make PauliSumOp.
PauliSumOp will support json https://github.com/Qiskit/qiskit-terra/pull/6119.

Reply:
I agree that huge unittests are no longer unittests but closer to integration tests. However, with some of these problems which we are getting into, there sometimes simply is no smaller test (which is still physical). Yes, one could do unphysical and purely numerical tests, but we should always have at least one physically reasonable test.

With that said, if you look at the `test/.../resources/file_parser.py`, he is actually constructing a `PauliSumOp` and not comparing purely strings here. I agree though, that some more structure file format for the resources data would be beneficial.

Reply:
I'm sorry that I did not noticed `file_parser.py`. Then No Problem.

The remaining problem I pointed out is replacement `assert` with `assertEqual`.